### ISSUE TITLE
[a11y]: NotificationsPanel: use proper heading levels

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/ibm-products#7173]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.62

### React version

18

### Automated testing tool and ruleset

accessibility-checker

### Description

As documented in https://www.w3.org/WAI/tutorials/page-structure/headings/, you shouldn't just jump from H2 to H6.   NotificationsPanel should be updated to use `Section` and `Heading` rather than hardcoded H2 and H6.  Starting with a `Section` (instead of a DIV) as the outermost element.

### WCAG 2.1 Violation

https://www.w3.org/TR/WAI-WEBCONTENT/wai-pageauth.html#tech-logical-headings

### Reproduction/example

https://carbon-for-ibm-products.netlify.app/?path=/story/ibm-products-components-notifications-panel-notificationspanel--default&globals=viewport:basic

### Steps to reproduce

Just check heading levels in Chrome developer tools.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by elycheea** [comment]
Thanks @wkeese for reporting all these. It looks like we even have `h6` set within the individual notifications. 🙈 We’ll move this to our refinement to discuss the appropriate heading levels or labels for this issue.
